torch/csrc/jit/passes/autocast.cpp

```diff
@@ -437,7 +444,9 @@ void handleBlock(Block* block, AutocastContext initial_state) {
 
       // Banned in autocast, see binary_cross_entropy_banned()
       case aten::binary_cross_entropy:
-        AT_ERROR("Unsafe to autocast");
+        if (has_seen_amp_section) {
+          AT_ERROR("Unsafe to autocast");
+        }
```

Comment:
is this a reasonable approach for handling binary_cross_entropy?

Reply:
or should we be checking `current_state()` to match eager autocasting mode behavior?

Reply:
This is already better than what we had :stuck_out_tongue: 

But it feels a little strange to me. I thought it's easier to just peek `current_state()` and assert if we don't want user to shoot at their own feet. Any specific reason that we want to assert when `binary_cross_entropy` is safely outside the amp enabled context?

Reply:
yeah, probably `current_state()` is fine and less likely to break anything that's already running, I'll go change that.